source/extensions/access_loggers/wasm/config.cc

```diff
@@ -20,30 +20,29 @@ namespace AccessLoggers {
 namespace Wasm {
 
 AccessLog::InstanceSharedPtr
-WasmAccessLogFactory::createAccessLogInstance(const Protobuf::Message& config,
+WasmAccessLogFactory::createAccessLogInstance(const Protobuf::Message& proto_config,
                                               AccessLog::FilterPtr&& filter,
                                               Server::Configuration::FactoryContext& context) {
-  const auto& wal_config =
-      MessageUtil::downcastAndValidate<const envoy::config::accesslog::v2::WasmAccessLog&>(config);
+  const auto& config =
+      MessageUtil::downcastAndValidate<const envoy::config::accesslog::v2::WasmAccessLog&>(
+          proto_config);
+  auto id = config.id();
+  auto configuration = std::make_shared<std::string>(config.configuration());
   auto tls_slot = context.threadLocal().allocateSlot();
-  if (wal_config.has_vm_config()) {
-    auto base_wasm =
-        Common::Wasm::createWasm(wal_config.id(), wal_config.vm_config(), context.api());
-    tls_slot->set([&wal_config, &base_wasm, &context](Event::Dispatcher& dispatcher) {
-      auto result =
-          Common::Wasm::createThreadLocalWasm(*base_wasm, wal_config.vm_config(), dispatcher,
-                                              wal_config.configuration(), context.api());
-      result->setClusterManager(context.clusterManager());
+  if (config.has_vm_config()) {
+    auto base_wasm = Common::Wasm::createWasm(id, config.vm_config(), context.clusterManager(),
+                                              context.dispatcher(), context.api());
+    // NB: the Slot set() call doesn't complete inline, so all arguments must outlive this call.
+    tls_slot->set([base_wasm, configuration](Event::Dispatcher& dispatcher) {
+      auto result = Common::Wasm::createThreadLocalWasm(*base_wasm, *configuration, dispatcher);
```

Comment:
This is coming in as a const&. If the configuration is large then it is more efficient to use std::shared_ptr and pass it in as a const& std::string.  Since this isn't on the critical path a couple extra cycles for the std::shared_ptr is a reasonable tradeoff.